### DISCUSSION TITLE
Zigbee2MQTT Dashboard in HA

### DISCUSSION THREAD

**Discussion by reefland** [Koenkk/zigbee2mqtt#10485] [Show and tell]
I'm trying to build a dashboard in HA that gives me a bit information about what's going on in Z2M.   I'm a novice at MQTT and parsing JSON, but I have a few items in place:

![zigbee2mqtt_dashboard](https://user-images.githubusercontent.com/16230914/147856436-66b8aa9a-0d5a-4617-a3ec-06f66fe666e5.png)

Here's what I used in Home Assistant's `sensors.yaml` file to create the sensors:

```
# Zigbee2MQTT Sensors
- platform: mqtt
  name: "Zigbee2MQTT Bridge Status"
  icon: mdi:clock-outline
  state_topic: "zigbee2mqtt/bridge/state"

- platform: mqtt
  name: "Zigbee2MQTT Verion"
  icon: mdi:numeric-1-box-multiple-outline
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.version }}"

- platform: mqtt
  name: "Zigbee2MQTT Coordinator Zigbee Channel"
  icon: mdi:numeric
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.network.channel }}"

- platform: mqtt
  name: "Zigbee2MQTT Coordinator Type"
  icon: mdi:card-text-outline
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.coordinator.type }}"

- platform: mqtt
  name: "Zigbee2MQTT Coordinator Firmware"
  icon: mdi:card-text-outline
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.coordinator.meta.revision }}"

- platform: mqtt
  name: "Zigbee2MQTT Device Count"
  icon: mdi:numeric
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.config.devices | count }}"

- platform: mqtt
  name: "Zigbee2MQTT Defined Groups"
  icon: mdi:numeric
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.config.groups | count }}"
```

Please share any other useful items you have been able to extract.

**Comment by trizmark** [comment]
I also extract the coordinator version:

```
- platform: mqtt
  name: Coordinator Version
  state_topic: "zigbee2mqtt/bridge/info"
  value_template: "{{ value_json.coordinator['meta']['majorrel'] }}.{{ value_json.coordinator['meta']['minorrel'] }}.{{ value_json.coordinator['meta']['maintrel'] }} [{{ value_json.coordinator['type'] }}]"
  icon: mdi:chip
```


This gives you something along the lines of: `2.7.1 [zStack3x0]`

**Reply by reefland** [reply]
Nice, added to my dashboard as well.

**Comment by reefland** [comment]
Another thought, trying to make an overall glance of the Zigbee network that Z2M is tracking.   I tried doing that dynamically parsing the Z2M topics but could not come up with a usable solution.  ZHA on Home Assistant has the frontend integration named `ZHA Network Card` -  which is perfect, a nice table view of the network with attributes you can customize.  I don't want a MAP view (only Z2M cards I found).

Forking that and adopting to Z2M is above my skills for now.  The low hanging fruit solution, was to use the `last_seen` attribute and then use the `Lovelace auto-entities` card which allows for wildcard entity selection:

```
  - type: custom:auto-entities
    card:
      type: glance
      title: Zigbee Devices
      show_icon: false
    filter:
      include:
        - entity_id: sensor.*_last_seen
    sort:
      method: state
      reverse: null
```
Since the `glances` card is being used to render, I get something reasonably compact but rather ugly to me:
![zigbee_devices_glances](https://user-images.githubusercontent.com/16230914/148095962-d005b1b0-424c-49b9-8e2b-e0c36e1b2ccb.png)
I haven't set the Friendly Names yet to be a little nicer.  Note that my devices which are members of Zigbee light groups don't seem to update the `last_seen` nor `link_quality` attributes.  Which I asked about here: #10551
